pkg/chargeback/http_test.go

```diff
@@ -616,3 +616,317 @@ func TestAPIV2ReportsGet(t *testing.T) {
 		})
 	}
 }
+
+//for v2 endpoints TableHidden
+func apiReportV2URLTable(reportName string) string {
+	return path.Join(APIV2Reports, reportName, "table")
+}
+
+func TestAPIV2ReportsTable(t *testing.T) {
+	const namespace = "default"
+	const testReportName = "test-report"
+	const testQueryName = "test-query"
+	reportStart := time.Time{}
+	reportEnd := reportStart.AddDate(0, 1, 0)
+
+	tests := map[string]struct {
+		reportStatus v1alpha1.ReportStatus
+		reportName   string
+
+		report      *v1alpha1.Report
+		query       *v1alpha1.ReportGenerationQuery
+		prestoTable *v1alpha1.PrestoTable
+
+		expectedStatusCode int
+		expectedAPIError   string
+
+		queryerPrepareFunc func(*mockpresto.MockExecQueryer) []presto.Row
+		expectedResults    *GetReportResults
+	}{
+		"report-finished-with-results-tableHidden-false": {
+			reportName: testReportName,
+			report:     newTestReport(testReportName, namespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{Phase: v1alpha1.ReportPhaseFinished}),
+			query: newTestReportGenQuery(testQueryName, namespace, []v1alpha1.ReportGenerationQueryColumn{
+				{
+					Name:        "timestamp",
+					Type:        "timestamp",
+					TableHidden: false,
+				},
+				{
+					Name:        "foo",
+					Type:        "double",
+					TableHidden: false,
+				},
+			},
+			),
+			prestoTable: newTestPrestoTable(testReportName, namespace, []hive.Column{
+				{
+					Name: "timestamp",
+					Type: "timestamp",
+				},
+				{
+					Name: "foo",
+					Type: "double",
+				},
+			},
+			),
+			queryerPrepareFunc: func(mock *mockpresto.MockExecQueryer) []presto.Row {
+				result := []presto.Row{
+					{
+						"timestamp": time.Time{},
+						"foo":       1,
+					},
+				}
+				mock.EXPECT().Query(gomock.Any()).Return(result, nil)
+				return result
+			},
+			expectedStatusCode: http.StatusOK,
+			expectedResults: &GetReportResults{
+				Results: []ReportResultEntry{
+					{
+						Values: []ReportResultValues{
+							{
+								Name:        "foo",
+								Value:       1,
+								TableHidden: false,
+							},
+						},
+					},
+				},
+			},
+		},
+		//add here
+		"report-finished-with-results-tableHidden-true": {
+			reportName: testReportName,
+			report:     newTestReport(testReportName, namespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{Phase: v1alpha1.ReportPhaseFinished}),
+			query: newTestReportGenQuery(testQueryName, namespace, []v1alpha1.ReportGenerationQueryColumn{
+				{
+					Name:        "timestamp",
+					Type:        "timestamp",
+					TableHidden: true,
+				},
+				{
+					Name:        "foo",
+					Type:        "double",
+					TableHidden: true,
```

Comment:
One of these should be false so we can see results with one column removed, and one kept in the results.

Reply:
My thinking was that the tableHidden-true test would have two true bools and tableHiddenFalse would test two false bools.  On the test on line 646 both bools are false.
Would you like each test (tableHidden-true and tableHidden-false) to have one of each bool then? For consistency? Ie change tableHidden false test as well.

Reply:
Sure that's fine.

Reply:
Done

Reply:
Actually now that both one of these test cases is redundant since it tests both behaviors, hidding and showing a column. Can you remove `report-finished-with-results-tableHidden-true` and rename `report-finished-with-results-tableHidden-false` to `report-finished-with-results`.

Reply:
Very true.
Done.